File: 283-move-zeroes/283-move-zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int l = 0;
        int r = 0;
        while(r<nums.size())
        {
            if(nums[r]==0){
                r++;
            }
            else
            {
                int tmp = nums[r];
                nums[r] = nums[l];
                nums[l] = tmp;            
                r++;
                l++;
            }
        }
    }
};

File: 283-move-zeroes/283-move-zeroes.java
class Solution {
    public void moveZeroes(int[] nums) {
        
        int swap=0;
        
        for(int i=0, j=0; i<nums.length-1 && j!=nums.length;j++) {
            if(nums[j]!=0) {
                swap=nums[i];
                nums[i]=nums[j];
                nums[j]=swap;
                i++;
            }
        }
    }
}